js/demo_policy/main.ts

```diff
@@ -0,0 +1,202 @@
+import type { Pod } from 'kubernetes-types/core/v1';
+
+import { Network } from '../kubewarden/host_capabilities/network';
+import { Validation } from '../kubewarden/validation';
+import { writeOutput } from '../protocol';
+import { ManifestDigest } from '../kubewarden/host_capabilities/oci/manifest_digest/manifest_digest';
+
+declare function policyAction(): string;
+
+interface PolicySettings {
+  ignoredNamespaces?: string[];
+  testScenario?: string;
+}
+
+class PolicySettings {
+  constructor(ignoredNamespaces: string[] = [], testScenario?: string) {
+    this.ignoredNamespaces = ignoredNamespaces;
+    this.testScenario = testScenario;
+  }
+}
+
+/**
+ * Handles OCI manifest digest lookup success scenario
+ */
+function handleOciManifestDigestSuccess(): Validation.ValidationResponse {
+  const image = 'ghcr.io/kubewarden/kubectl:v1.31.0';
+  let digest: string | null = null;
+  try {
+    digest = ManifestDigest.getOCIManifestDigest(image);
+  } catch (e) {
+    console.error('OCI manifest digest lookup failed:', e);
+  }
+  return new Validation.ValidationResponse(
+    !!digest,
+    digest ? undefined : 'Failed to retrieve OCI manifest digest',
+    undefined,
+    undefined,
+    { digest: digest || '' }
+  );
+}
+
+/**
+ * Handles OCI manifest digest lookup failure scenario
+ */
+function handleOciManifestDigestFailure(): Validation.ValidationResponse {
+  const image = 'registry.testing.lan/nonexistent-image:1.0.0';
+  let digest: string | null = null;
+  try {
+    digest = ManifestDigest.getOCIManifestDigest(image);
+  } catch (e) {
+    console.error('OCI manifest digest lookup failed:', e);
+    return new Validation.ValidationResponse(
+      false,
```

Comment:
this is a bit counter intuitive, but here I would actually mark this a `true`, since we actually wanted to host to throw an error.

Generally speaking I would say: if the test scenario is doing what we expect -> the validation response should be type accepted, no matter whether the was a failure or not